app/src/main/kotlin/org/kotlinlsp/index/worker/IndexKtFile.kt

```diff
@@ -52,7 +50,117 @@ fun indexKtFile(project: Project, ktFile: KtFile, db: Database) {
             val startOffset = dcl.textOffset
             val endOffset = dcl.textOffset + name.length
 
-            db.putDeclarationForFile(fileRecord.path, name, startOffset, endOffset)
+            val decl = project.read {
+                analyze(dcl) {
+                    when (dcl) {
```

Comment:
This `when` will become bigger and bigger, at least for now let's move it outside the `indexKtFile` method